### PR TITLE
fix: return error from subprocesses

### PR DESCRIPTION
**Pull request by atanmarko**
# Description

Return topos cli exit status with respect to edge subprocess status

Fixes TP-825

## PR Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added or updated tests that comprehensively prove my change is effective or that my feature works

### CODE REVIEW

**Comment by atanmarko** [comment]
Somehow polygon edge spawned process always returns `ExitStatus(unix_wait_status(0))` despite the fact that error happened. I have tried other commands (`ls`) and their error is correctly propagated to `topos cli` exit

File: crates/topos/src/components/node/mod.rs [hunk 1]
```diff
@@ -79,13 +79,30 @@ pub(crate) async fn handle_command(
...
+                    }
+                    Ok(Err(e)) => {
+                        println!("Failed to generate edge config with error {e}");
+                        remove_dir_all(node_path).expect("failed to remove config folder");
+                        std::process::exit(1);
+                    }
+                    Err(e) => {
                         println!("Failed to generate edge config");
                         remove_dir_all(node_path).expect("failed to remove config folder");
                         std::process::exit(1);
```
Comment (Freyskeyd) [inline crates/topos/src/components/node/mod.rs:108]:
```suggestion
                    Ok(Err(e)) | Err(e) => {
                        println!("Failed to generate edge config with error {e}");
                        remove_dir_all(node_path).expect("failed to remove config folder");
                        std::process::exit(1);
```

I think you can simplify this with that

File: crates/topos/src/components/node/mod.rs [hunk 2]
```diff
@@ -253,12 +270,26 @@ pub(crate) async fn handle_command(
...
+                    match result {
+                        Ok(Ok(status)) => {
+                            if let Some(0) = status.code() {
+                                info!("Terminating with success error code");
+                            } else {
+                                info!("Terminating with error status: {:?}", status);
+                                std::process::exit(1);
+                            }
+                        }
+                        Ok(Err(e)) => {
```
Comment (Freyskeyd) [inline crates/topos/src/components/node/mod.rs:284]: Same here for the shorter version

File: crates/topos/src/edge/mod.rs [hunk 3]
```diff
@@ -99,8 +100,18 @@ impl CommandConfig {
 
         let (running_out, _) = tokio::join!(running, logging);
 
-        info!("The Edge process is terminated");
-        running_out
+        let exit_status = running_out?;
+
+        info!(
```
Comment (Freyskeyd) [inline crates/topos/src/edge/mod.rs:105]: That's a lot of information, let's keep that like that and see later if it doesn't display too much things.

**Review by Freyskeyd — APPROVED** [review]

File: crates/topos/src/components/node/mod.rs (hunk 1 above)
Comment (atanmarko) [reply crates/topos/src/components/node/mod.rs:108]: Tried it already, compiler gives:
```
 this expression has type `std::result::Result<std::result::Result<std::process::ExitStatus, components::node::services::process::Errors>, tokio::task::JoinError>`
...
100 |                       Ok(Err(e)) |  Err(e) => {
    |                              -          ^ expected `Errors`, found `JoinError`
```

File: crates/topos/src/components/node/mod.rs (hunk 2 above)
Comment (atanmarko) [reply crates/topos/src/components/node/mod.rs:284]: Same compiler error as before - error must be of the same type to match such a pattern

File: crates/topos/src/components/node/mod.rs (hunk 1 above)
Comment (Freyskeyd) [reply crates/topos/src/components/node/mod.rs:108]: Ah it is a `JoinError` ?
Maybe you can remove the `JoinHandle` and transform `generate_edge_config` into an `async` function.

**Pull request merged**